service/matching/matching_engine.go

```diff
@@ -371,60 +371,129 @@ func (e *matchingEngineImpl) getTaskQueuePartitionManager(
 	create bool,
 	loadCause loadCause,
 ) (taskQueuePartitionManager, error) {
-	pm, err := e.getTaskQueuePartitionManagerNoWait(partition, create, loadCause)
+	key := partition.Key()
+
+	e.partitionsLock.RLock()
+	pm, ok := e.partitions[key]
+	e.partitionsLock.RUnlock()
+
+	if ok {
+		return pm, nil
+	}
+
+	if !create {
+		return nil, nil
+	}
+
+	pm, created, err := e.createTaskQueuePartitionManager(partition, loadCause)
 	if err != nil || pm == nil {
 		return nil, err
 	}
+
+	if pm.Partition().IsRoot() && created {
+		// Whenever a root partition is loaded we need to force all other partitions to load.
+		// If there is a backlog of tasks on any child partitions force loading will ensure that they
+		// can forward their tasks the poller which caused the root partition to be loaded.
+		// These partitions could be managed by this matchingEngineImpl, but are most likely not.
+		// We skip checking and just make gRPC requests to force loading them all.
+		err := e.forceLoadAllNonRootPartitions(ctx, partition)
+		if err != nil {
```

Comment:
the individual rpc calls already log stuff, no need to do it here (this call doesn't return an error anyway)

Reply:
Could you be thinking this is a call to the RPC? This calls the function right below which does return an error.

Reply:
Ahh, I see what you're saying now? How would you propose the code be improved? 

This is how I see possible solutions:
1. **Callee swallows the error - Deceptive function signature, provides a bad contract to caller**; Could result in silent failures we can't predict if the implementation of namespace lookups changes or fails.
2. **Caller swallows the error - Doesn't fulfill the contract outlined by callee**; Same problem as the first, plus now a future dev may change forceLoadAllNonRootPartitions to return additional error cases and assume that the caller handles errors because it was outlined in the contract.

Reply:
This concern went away after moving ForceLoadAllNonRootPartitions into partitionManager